packages/server/graphql/mutations/addPokerTemplateScale.ts

```diff
@@ -0,0 +1,102 @@
+import {GraphQLID, GraphQLNonNull} from 'graphql'
+import {SubscriptionChannel, Threshold} from 'parabol-client/types/constEnums'
+import dndNoise from 'parabol-client/utils/dndNoise'
+import getRethink from '../../database/rethinkDriver'
+import TemplateScale from '../../database/types/TemplateScale'
+import {getUserId, isTeamMember} from '../../utils/authorization'
+import publish from '../../utils/publish'
+import standardError from '../../utils/standardError'
+import AddPokerTemplateScalePayload from '../types/AddPokerTemplateScalePayload'
+
+const addPokerTemplateScale = {
+  description: 'Add a new scale for the poker template',
+  type: new GraphQLNonNull(AddPokerTemplateScalePayload),
+  args: {
+    parentScaleId: {
+      type: GraphQLID
+    },
+    teamId: {
+      type: new GraphQLNonNull(GraphQLID)
+    }
+  },
+  async resolve(_source, {parentScaleId, teamId}, {authToken, dataLoader, socketId: mutatorId}) {
+    const r = await getRethink()
+    const operationId = dataLoader.share()
+    const subOptions = {operationId, mutatorId}
+    const viewerId = getUserId(authToken)
+
+    // AUTH
+    if (!isTeamMember(authToken, teamId)) {
+      return standardError(new Error('Team not found'), {userId: viewerId})
+    }
+
+    // VALIDATION
+    const activeScales = await r
+      .table('TemplateScale')
+      .getAll(teamId, {index: 'teamId'})
+      .filter((row) =>
+        row('removedAt')
+          .default(null)
+          .eq(null)
+      )
+      .run()
+    if (activeScales.length >= Threshold.MAX_POKER_TEMPLDATE_SCALES) {
+      return standardError(new Error('Too many scales'), {userId: viewerId})
+    }
+
+    // RESOLUTION
+    let newScale
+    const sortOrder = Math.max(0, ...activeScales.map((scale) => scale.sortOrder)) + 1 + dndNoise()
+    if (parentScaleId) {
+      const parentScale = (await dataLoader
+        .get('TemplateScales')
+        .load(parentScaleId)) as TemplateScale
+      if (!parentScale) {
+        return standardError(new Error('Parent scale not found'), {userId: viewerId})
+      }
+      if (!!parentScale.isStarter && parentScale.teamId !== teamId) {
```

Comment:
need to remove a `!`?

Reply:
Actually, meant to be triple ! to handle both `false` and `undefined`